lms/views/basic_lti_launch.py

```diff
@@ -112,18 +111,16 @@ def canvas_file_basic_lti_launch(self):
         # Canvas skips that step. We are doing this to ensure that there is a
         # module item configuration. As a result of this we can rely on this
         # being around in future code.
+        document_url = f"canvas://file/course/{course_id}/file_id/{file_id}"
         self.assignment_service.set_document_url(
             self.request.params["tool_consumer_instance_guid"],
             resource_link_id,
             # This URL is mostly for show. We just want to ensure that a module
             # configuration exists. If we're going to do that we might as well
             # make sure this URL is meaningful.
-            document_url=f"canvas://file/course/{course_id}/file_id/{file_id}",
+            document_url=document_url,
         )
-
-        self.context.js_config.add_canvas_file_id(course_id, resource_link_id, file_id)
-
-        return self.basic_lti_launch(grading_supported=False)
+        return self.basic_lti_launch(document_url=document_url, grading_supported=False)
```

Comment:
Ok, so previously `canvas_file_basic_lti_launch()` called `js_config.add_canvas_file_id()` itself then called `self.basic_lti_launch()` with `document_url=None` which meant that `self.basic_lti_launch()` *did not* call `js_config.add_document_url()`. So `js_config.add_canvas_file_id()` was called and `js_config.add_document_url()` was not called.

Now `canvas_file_basic_lti_launch()` does not call `js_config.add_canvas_file_id()` anymore but instead calls `self.basic_lti_launch()` with a `document_url` so `self.basic_lti_launch()` *will* now call `js_config.add_document_url()`. So it has swapped: now `js_config.add_canvas_file_id()` is no longer called but `js_config.add_document_url()` *is* called.

This sounds like it breaks things: does `canvas_file_basic_lti_launch()` now add a document URL instead of a Canvas file ID? But I see that you've added `canvas://file/` URL special-casing to `JSConfig.add_document_url()` so that calling `add_document_url()` with a `canvas://file/` does the same as what calling `add_canvas_file_id()` would do. So actually there should not be any functional change here.

I really don't think the `basic_lti_launch()` method is helping to make the code easy to understand here. I get that it's trying to deduplicate code. But I think once it contains optional behaviour like `document_url` and `grading_supported` it just makes things hard to follow. It's kind of classic procedural spaghetti code to be honest. So I'm glad to see `document_url` becoming no longer optional.